lib/block-supports/typography.php

```diff
@@ -77,6 +82,23 @@ function gutenberg_apply_typography_support( $block_type, $block_attributes ) {
 		}
 	}
 
+	// Font Family.
+	if ( $has_font_family_support ) {
+		$has_font_family = isset( $block_attributes['style']['typography']['fontFamily'] );
+		// Apply required class and style.
+		if ( $has_font_family ) {
+			$font_family = $block_attributes['style']['typography']['fontFamily'];
+			if ( strpos( $font_family, 'var:preset|font-family' ) !== false ) {
+				// Get the name from the string and add proper styles.
+				$index_to_splice  = strrpos( $font_family, '|' ) + 1;
+				$font_family_name = substr( $font_family, $index_to_splice );
+				$styles[]         = sprintf( 'font-family: var(--wp--preset--font-family--%s);', $font_family_name );
+			} else {
+				$styles[] = sprintf( 'font-family: %s;', $block_attributes['style']['color']['fontFamily'] );
```

Comment:
Under which situations a font-family value is not in the form of `var:preset|font-family|etc`?

Unlike other style properties, users can't provide custom values, so I don't see how this path is reached. I presume we can remove it. The fact that this has been using `style.COLOR.fontFamily` without a bug being reported seems to confirm this is a code path we don't need. cc @jorgefilipecosta @aristath 

For context, this comes from a PR I'm working on at https://github.com/WordPress/gutenberg/pull/31910 to use classes instead of CSS Custom Properties for font-family.

Reply:
I guess a situation where font-family may not be preset is in the case of a block pattern where the pattern is excitedly using a specific font family in a block.

Regarding the usage of classes for font-family presets I think we can use variables for now it is not clear classes bring any advantage in that use case. For the colors, we are outputting CSS vars anyway so a block could use the vars directly and we need to support that. For the font family, I guess we can keep with just vars for now and see what feedback we get.

Reply:
> I guess a situation where font-family may not be preset is in the case of a block pattern

Nice thinking, thanks! I'll keep that use case then.

> Regarding the usage of classes for font-family presets I think we can use variables for now it is not clear classes bring any advantage in that use case

I'd think we want to do it the other way around: what's the advantage of font-family using custom properties instead of classes, as the other style properties do?